pkg/_pub_shared/lib/search/search_form.dart

```diff
@@ -18,6 +18,7 @@ final _sortRegExp = RegExp('sort:([a-z]+)');
 final _updatedRegExp = RegExp('updated:([0-9][0-9a-z]*)');
 final _tagRegExp =
     RegExp(r'([\+|\-]?[a-z0-9]+:[a-z0-9\-_\.]+)', caseSensitive: false);
+final _topicRegExp = RegExp(r'([\+|\-])?#([a-z0-9-]{2,32})');
```

Comment:
Is '|' meant to be an option for the sign?
```
void main() {
  final _topicRegExp = RegExp(r'([\+|\-])?#([a-z0-9-]{2,32})');
  print(_topicRegExp.firstMatch(r'|#21')?[1] );
}
```
prints '|'

Reply:
good catch, we have the same bug in `_tagRegExp`, fixing both.